### PR TITLE
Expose concurrency flag of envoy

### PR DESCRIPTION
**Pull request by mcieplak**
This is to address the following github issue: https://github.com/istio/istio/issues/3031

### CODE REVIEW

**Comment by mcieplak** [comment]
This PR is being used here: https://github.com/istio/istio/pull/3065

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -109,6 +109,9 @@ message ProxyConfig {
   // Envoy's internal metrics take up 67 characters, for a total of 256 character name per metric.
   // Increase the value of this field if you find that the metrics from Envoys are truncated.
   int32 stat_name_length = 15;
+
+  // The # of worker threads to run. Default value is number of cores on the machine.
```
Comment (ZackButcher) [inline mesh/v1alpha1/config.proto]: nit: write out "number" please.

Comment (mcieplak) [reply mesh/v1alpha1/config.proto]: @ZackButcher Sure. Updated.

**Comment by mcieplak** [comment]
@ZackButcher @rshriram Thank you for the reviews. Can we please mark this PR as ok to test?

**Comment by ZackButcher** [comment]
/ok-to-test

Whoops, sorry, forgot we had to on this repo.

**Comment by mcieplak** [comment]
@ZackButcher Thank you! Build is passing now, can we please merge?

**Pull request merged**